src/lib/nvme_pt_ioctl.c

```diff
@@ -949,13 +952,19 @@ static struct opal_req_item opal_generic_get_column_cmd[] = {
 static int opal_generic_get_column(int fd, struct opal_device *dev, const uint8_t *uid,
 				   uint64_t start_col, uint64_t end_col)
 {
+	int ret;
+
 	opal_generic_get_column_cmd[3].val.uint = start_col;
 	opal_generic_get_column_cmd[7].val.uint = end_col;
 
 	prepare_req_buf(dev, opal_generic_get_column_cmd, ARRAY_SIZE(opal_generic_get_column_cmd),
 			uid, opal_method[OPAL_GET_METHOD_UID]);
 
-	return opal_snd_rcv_cmd_parse_chk(fd, dev, false);
+	ret = opal_snd_rcv_cmd_parse_chk(fd, dev, false);
+	if (ret)
```

Comment:
why tokens are not freed when opal_snd_rcv_cmd_parse_chk returns success (status =0)?

Reply:
the content of the payload is utilized in other functions like, opal_get_msid, opal_get_lsp_lifecycle, get_table_length, list_lr and a few more. If we put all the tokens down here, these functions can loose that payload value.

Reply:
yep. I noticed that. and here is some concern: before this change the contract was that payload must be freed in the caller. With this change in some cases caller frees the payload in some other cases it is freed in this function. It is really confusing and error prone

Reply:
Only in case of success, the caller function will put_all_token.
In case of error, the called func will take care of it